src/redux/explorer.ts

```diff
@@ -0,0 +1,1076 @@
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_TRANSACTIONS.RECEIVED_OPTIMISM,
+    (message: TransactionsReceivedMessage) => {
+      // logger.log('optimism txns received', message?.payload?.transactions);
+      dispatch(transactionsReceived(message));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_TRANSACTIONS.RECEIVED_POLYGON,
+    (message: TransactionsReceivedMessage) => {
+      // logger.log('polygon txns received', message?.payload?.transactions);
+      dispatch(transactionsReceived(message));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_TRANSACTIONS.APPENDED,
+    (message: TransactionsReceivedMessage) => {
+      logger.log('txns appended', message?.payload?.transactions);
+      dispatch(transactionsReceived(message, true));
+      // Fetch balances onchain to override zerion's
+      // which is likely behind
+      dispatch(fetchOnchainBalances({ keepPolling: false, withPrices: false }));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_TRANSACTIONS.CHANGED,
+    (message: TransactionsReceivedMessage) => {
+      logger.log('txns changed', message?.payload?.transactions);
+      dispatch(transactionsReceived(message, true));
+      // Fetch balances onchain to override zerion's
+      // which is likely behind
+      dispatch(fetchOnchainBalances({ keepPolling: false, withPrices: false }));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_TRANSACTIONS.REMOVED,
+    (message: TransactionsRemovedMessage) => {
+      logger.log('txns removed', message?.payload?.transactions);
+      dispatch(transactionsRemoved(message));
+      // Fetch balances onchain to override zerion's
+      // which is likely behind
+      dispatch(fetchOnchainBalances({ keepPolling: false, withPrices: false }));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_ASSETS.RECEIVED_ARBITRUM,
+    (message: L2AddressAssetsReceivedMessage) => {
+      dispatch(l2AddressAssetsReceived(message, Network.arbitrum));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_ASSETS.RECEIVED_OPTIMISM,
+    (message: L2AddressAssetsReceivedMessage) => {
+      dispatch(l2AddressAssetsReceived(message, Network.optimism));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_ASSETS.RECEIVED_POLYGON,
+    (message: L2AddressAssetsReceivedMessage) => {
+      dispatch(l2AddressAssetsReceived(message, Network.polygon));
+    }
+  );
+
+  socket.on(
+    messages.ADDRESS_ASSETS.RECEIVED,
+    (message: AddressAssetsReceivedMessage) => {
+      dispatch(addressAssetsReceived(message));
+      if (isValidAssetsResponseFromZerion(message)) {
+        logger.log(
+          '😬 Cancelling fallback data provider listener. Zerion is good!'
+        );
+        dispatch(disableFallbackIfNeeded());
+        dispatch(explorerInitL2(Network.optimism));
+        // Fetch balances onchain to override zerion's
+        // which is likely behind
+        dispatch(
+          fetchOnchainBalances({ keepPolling: false, withPrices: false })
```

Comment:
i built the app and this is being triggered, i don't see it on develop

https://user-images.githubusercontent.com/12115171/172211063-a7f487dd-0c61-4313-ac67-dcbee332caae.mov

Reply:
Hmm that's interesting, I'll look into it ASAP. Thanks for catching it!

Reply:
Should be fixed now, thanks for the heads up!

Reply:
cool, taking a look again